analytic_engine/src/meta/details.rs

```diff
@@ -217,61 +235,79 @@ pub struct ManifestImpl {
 }
 
 impl ManifestImpl {
-    pub async fn open(wal_manager: WalManagerRef, opts: Options) -> Result<Self> {
+    pub async fn open(
+        opts: Options,
+        wal_manager: WalManagerRef,
+        store: ObjectStoreRef,
+    ) -> Result<Self> {
         let manifest = Self {
             opts,
             wal_manager,
+            store,
             num_updates_since_snapshot: Arc::new(AtomicUsize::new(0)),
             snapshot_write_guard: Arc::new(Mutex::new(())),
         };
 
         Ok(manifest)
     }
 
-    async fn store_update_to_wal(&self, request: MetaUpdateRequest) -> Result<SequenceNumber> {
-        info!("Manifest store update, request:{:?}", request);
-
-        let payload: MetaUpdatePayload = MetaUpdateLogEntry::Normal(request.meta_update).into();
-        let log_batch_encoder =
-            self.wal_manager
-                .encoder(request.location)
-                .context(GetLogBatchEncoder {
-                    wal_location: request.location,
-                })?;
-        let log_batch = log_batch_encoder.encode(&payload).context(EncodePayloads {
-            wal_location: request.location,
-        })?;
-
-        let store_ctx = self.opts.store_context();
-        let write_ctx = WriteContext {
-            timeout: store_ctx.timeout,
+    async fn store_update_to_wal(
+        &self,
+        meta_update: MetaUpdate,
+        location: WalLocation,
+    ) -> Result<SequenceNumber> {
+        let log_store = WalBasedLogStore {
+            opts: self.opts.clone(),
+            location,
+            wal_manager: self.wal_manager.clone(),
         };
-
-        self.wal_manager
-            .write(&write_ctx, &log_batch)
-            .await
-            .context(WriteWal)
+        log_store.append(meta_update).await
     }
 
     /// Do snapshot if no other snapshot is triggered.
     ///
     /// Returns the latest snapshot if snapshot is done.
-    async fn maybe_do_snapshot(&self, location: WalLocation) -> Result<Option<Snapshot>> {
+    async fn maybe_do_snapshot(
```

Comment:
I find return value of this function is never used, remove it in this PR?

Reply:
It is used in the L316.